### PR TITLE
Additional methods for filling up the settings.

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

File: client/provider/web3-provider.js [hunk 1]
```diff
@@ -30,6 +36,7 @@ export class Web3Provider extends Web3Base {
       }
       return toPromise(obj.method).then(obj.actionMaker)
         .catch(err => {
+          console.error(err);
```
Comment (Adam-Goldman7) [inline client/provider/web3-provider.js]: why we need this?
we will c the error dispatched right?

Comment (tomusdrw) [inline client/provider/web3-provider.js]: But this gives you prety nice stack trace in DevTools - you can click the filename and will take you directly to the line that has errored. In action we just get a string.

**Comment by tomusdrw** [comment]
@Adam-Goldman7 There is some problem with e2e tests (we should increase the timeouts probably?)

File: client/provider/web3-provider.js (hunk 1 above)
Comment (Adam-Goldman7) [inline client/provider/web3-provider.js]: :+1:

**Comment by Adam-Goldman7** [comment]
@tomusdrw 
Im looking into it
there's an error on this branch

**Pull request merged**